packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -520,8 +539,9 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
 
   @override
   void hitTestInView(HitTestResult result, Offset position, int viewId) {
-    assert(viewId == renderView.flutterView.viewId);
-    renderView.hitTest(result, position: position);
+    final RenderView targetView = _viewIdToRenderView(viewId)!;
```

Comment:
It's either extracting it to a method `_viewIdToRenderView` or inlining the function here, because now that `hitTestInView` has a `viewId`, I have to check whether this ID is the implicit view ID, and skip the hit testing if the view is not the implicit view. I'm leaning to keep the current structure, and when you implement it with a map you can change the line to `final RenderView targetView = _viewIdToRenderView[viewId!;`. What do you think?

Reply:
There is currently no way this can give you anything but that ID, right? So I would argue that just firing an assert here is the right thing to do if the viewID is not the expected (and only valid) one. In the current implementation it would actually give you a null-dereferencing error on line 542 if we were to receive an invalid ID, which I think is worse then firing an assert clearly stating what the current expected behavior is.

Reply:
You're right! I'll do it.